server/repos/reposController.js

```diff
@@ -0,0 +1,64 @@
+var url = require('url');
+var utils = require('./reposUtils');
+var Promise = require('bluebird')
+var request = require('request');
+
+module.exports = {
+  getRepo: function(req, res) {
+    var href = url.parse(req.url);
+    var pathname = href.pathname.split('/');
+    var query = href.query;
+    var user = pathname[2];
+    var repo = pathname[4];
+
+    var R = Promise.promisify(utils.retrieveRepos);
```

Comment:
also getRepo and getRepos look very similar/share some code...it seems common in apis get all the elems (repos) and then if we provide an id we get that specific elem (repo)--can we copy their conventions instead of code repeat